### DISCUSSION TITLE
Complete Chinese and English documents advanced_guides/structures.md 完成中英文文档 advanced_guides/structures.md

### DISCUSSION THREAD

**Discussion by OpenMMLab-Assistant-004** [open-mmlab/OpenMMLabCamp#552] [MMagic]
# Complete Chinese and English documents advanced_guides/structures.md 
 完成中英文文档 advanced_guides/structures.md 

Complete Chinese and English documents [https://github.com/open-mmlab/mmagic/blob/main/docs/en/advanced_guides/structures.md](https://github.com/open-mmlab/mmagic/blob/main/docs/en/advanced_guides/structures.md) 
and [https://github.com/open-mmlab/mmagic/blob/main/docs/zh_cn/advanced_guides/structures.md](https://github.com/open-mmlab/mmagic/blob/main/docs/zh_cn/advanced_guides/structures.md) 

完成中英文文档 [https://github.com/open-mmlab/mmagic/blob/main/docs/en/advanced_guides/structures.md](https://github.com/open-mmlab/mmagic/blob/main/docs/en/advanced_guides/structures.md) 
以及 [https://github.com/open-mmlab/mmagic/blob/main/docs/zh_cn/advanced_guides/structures.md](https://github.com/open-mmlab/mmagic/blob/main/docs/zh_cn/advanced_guides/structures.md) 

## Task Information

**Repository**：MMagic

**Difficulty**：Easy

**Points**：20

**Estimated Time Consumption**：7 days

## Task Status

**Claimant's Github ID**：MING-ZCH

**Claim Time**：2023年08月08日

**PR Link**：https://github.com/open-mmlab/mmagic/pull/1982

## Points to Note

❗️请注意：此任务属于超级视客营任务，不能被领取！请登陆 [超级视客营平台](https://openmmlab.com/activity/codecamp) 按照流程申请任务，我们会尽快审核～

**Comment by MING-ZCH** [comment]
想问问这个数据结构文档主要是写哪些部分，有参考样例吗？希望给小白解惑

**Reply by zengyh1900** [reply]
hi @MING-ZCH ,
1. mmagic 中所使用的数据结构主要是 `DataSample`, 源码在 https://github.com/open-mmlab/mmagic/blob/main/mmagic/structures/data_sample.py#L74， 可以参考其中的 docstring
2. 整体上，mmagic 的 `DataSample` 继承自 mmengine 的 `BaseDataElement`，mmengine 中的 data structure 的介绍可以参考  https://mmengine.readthedocs.io/en/latest/advanced_tutorials/data_element.html
3. 同时也可以参考其他算法库，例如 mmocr： https://mmocr.readthedocs.io/en/latest/basic_concepts/structures.html

**Comment by MING-ZCH** [comment]
https://github.com/open-mmlab/mmagic/pull/1982
